src/Squirrel/IUpdateManager.cs

```diff
@@ -167,8 +167,11 @@ public static async Task<ReleaseEntry> UpdateApp(this IUpdateManager This, Actio
                 await This.ErrorIfThrows(() => 
                     This.CreateUninstallerRegistryEntry(),
                     "Failed to set up uninstaller");
-            } catch {
-                if (ignoreDeltaUpdates == false) {
+            } catch (IOException ex) when ((ex.HResult & 0xFFFF) == 0x27 || (ex.HResult & 0xFFFF) == 0x70) {
+                    throw ex; // Rethrow not enough space exception.
```

Comment:
Is it enough to handle both delta and full update memory exceptions?

Reply:
This try catch is over all updating process, so it should be enough. I'm not completely sure if all functions in .NET throws not enough space exception when running out of disk space. In that case full download tries to start and then fails again and rethrows the exception to our updater. This is also the case when there is not enough space while downloading package because there is only one generic exception in the DownloadManager.